doc/architecture/decisions/0014-p2p-ipc-abstraction.md

```diff
@@ -8,17 +8,17 @@ Proposed
 
 ## Context
 
-It's frustrating to admit, but I'm still pretty slow at writing rust code. Oh, I can hack something together quickly (gogo `unwrap()/clone()`), but when I sit down to write efficient production code, I get bogged down in cyclic lifetime reference issues or any number of other arcane rustisisms.
+Rust as a language, in my opinion, does not lend itself to rapid prototyping. Writing efficient Rust code is a problem domain unto itself. E.g. Is it worth returning borrowed references in this case to save memory? Am I able to craft the lifetimes appropriately given all the other api usages of my struct? What about synchronization? These questions are implementation details, not architectural. If we are still working out the details with the underlying architecture, we don't want to be spending brainpower on * how * we accomplish our experiments.
```

Comment:
Rust code is no less efficient than C/C++ code. Do you mean that writing Rust code efficiently is difficult?